File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let boats = 0;
    people.sort((a, b) => b - a); // 내림차순
    
    let start = 0; // 가장 무거운 사람
    let end = people.length - 1; // 가장 가벼운 사람
    
    while(start < end) { // 등호가 없는 이유는 등호가 있으면 한사람을 두명 태우는 꼴이 될 수도 있으므로
        if(people[start] + people[end] <= limit) {
            boats++; // 제일 가볍고, 무거운 사람을 태웠다.
            end-=1; 
            start+=1;
        }
        else {
            boats++; // 제일 무거운 사람과 같이 탈 사람이 없으므로 이 사람만 태워 보낸다.
            start+=1;
        }
    }
    if(start === end) boats++;
    return boats;
    
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    sorted_people = sorted(people)
    q = deque(sorted_people) # 오름차순
    count = 0
    
    while (True):
        if len(q) <= 0:
            return count
        
        if len(q) == 1:
            count += 1
            return count
        
        min_w = q[0]
        max_w = q[-1]
        
        if max_w + min_w <= limit:
            q.pop()
            q.popleft()
        else:
            q.pop()
            
        count += 1
        
        
    